File: judge/python/tests/two_difference.py
from common import *
from solution import *
import copy
import sys
import datetime

num_test = 50
true, false = True, False
in_0 = []
in_org_0 = []
in_1 = []
in_org_1 = []
out = []
def test(answer, i):
    if len(out[i]) != len(answer): return False
    if out[i][0] == -1 and out[i][1] == -1:
        return answer[0] == -1 and answer[1] == -1
    else:
        if answer[0] == -1 or answer[1] == -1:
            return False
        else:
            return abs(in_0[i][answer[0]] - in_0[i][answer[1]]) == in_1[i]
		

def load_test():
    f = open('judge/tests/two-difference.txt', 'r')
    global in_0, in_org_0
    in_0 = read_int_matrix(f)
    in_org_0 = copy.deepcopy(in_0)
    global in_1, in_org_1
    in_1 = read_int_array(f)
    in_org_1 = copy.deepcopy(in_1)
    global out
    out = read_int_matrix(f)
    f.close

def judge():
    load_test()
    start_time = datetime.datetime.now()
    for i in range(num_test):
        answer = two_dif(in_0[i], in_1[i]) 
        if (not test(answer, i)):
            out_str = str(i+1) + " / " + str(num_test) + ";"
            out_str += str(in_org_0[i])
            out_str += ", "
            out_str += str(in_org_1[i])
            out_str += ";"
            out_str += str(answer)
            out_str += ";"
            out_str += str(out[i])
            print(out_str)
            return

    delta = datetime.datetime.now() - start_time
    runtime = str(int(delta.total_seconds() * 1000))
    print('Accepted;' + runtime)

File: judge/ruby/tests/two-difference.rb
@num_test = 50
@in_0 = []
@in_org_0 = []
@in_1 = []
@in_org_1 = []
@out = []

def test(answer, i)
    if @out[i].length != answer.length
        return false
    end
    if @out[i][0] == -1 && @out[i][1] == -1
        return answer[0] == -1 && answer[1] == -1
    else
        if answer[0] == -1 || answer[1] == -1
            return false
        else
            return (@in_0[i][answer[0]] - @in_0[i][answer[1]]).abs == @in_1[i]
        end
    end
end

def load_test
    f = File.new("judge/tests/two-difference.txt")
    @in_0 = read_int_matrix(f)
    @in_org_0 = @in_0.dup
    @in_1 = read_int_array(f)
    @in_org_1 = @in_1.dup
    @out = read_int_matrix(f)
    f.close
end

def judge
    load_test

    start_time = Time.now

    (0...@num_test).each do |i|
        answer = two_dif(@in_0[i], @in_1[i]) 
        if (!test(answer, i))
            print "#{i+1} / #{@num_test};"
            print @in_org_0[i]
            print ', '
            print @in_org_1[i]
            print ';'
            print answer
            print ';'
            print @out[i]
            puts
            return
        end
    end

    runtime = (Time.now - start_time) * 1000.0
    puts('Accepted;' + runtime.to_i.to_s)
end
